runtime/ops/net.rs

```diff
@@ -138,18 +135,14 @@ async fn receive_udp(
 
   let rid = args.rid as u32;
 
-  let receive_fut = poll_fn(|cx| {
-    let mut state = state.borrow_mut();
-    let resource = state
-      .resource_table
-      .get_mut::<UdpSocketResource>(rid)
-      .ok_or_else(|| bad_resource("Socket has been closed"))?;
-    let socket = &mut resource.socket;
-    socket
-      .poll_recv_from(cx, &mut zero_copy)
-      .map_err(AnyError::from)
-  });
-  let (size, remote_addr) = receive_fut.await?;
+  let resource = state
+    .borrow_mut()
```

Comment:
Looks like https://github.com/denoland/deno/pull/8512/files/8fcb9a496bbadc19dc1c22c226ff17656b286bce#diff-6b4e31d40f717632f8b6eb94b2a0828d3cde3ee6df5197d49e976068f9be33b2 hasn't been addressed.
We also need a test for it, which should roughly do the following:

* Create an UDP socket.
* Start receiving packets on that UDP socket.
* Use the UDP socket to send a packet to itself.
* Verify that the packet has been received.

Reply:
test:

```ts
const socket = Deno.listenDatagram({ port: 3500, transport: "udp" });
assert(socket.addr.transport === "udp");
assertEquals(socket.addr.port, 3500);
assertEquals(socket.addr.hostname, "127.0.0.1");

const recvPromise = socket.receive();

const sendBuf = new Uint8Array([1, 2, 3]);
const sendLen = await socket.send(sendBuf, socket.addr);
assertEquals(sendLen, 3);

const [recvBuf, recvAddr] = await recvPromise;
assertEquals(recvBuf.length, 3);
assertEquals(1, recvBuf[0]);
assertEquals(2, recvBuf[1]);
assertEquals(3, recvBuf[2]);

socket.close();
```